### PR TITLE
fix: Update scam token warning

### PR DESCRIPTION
**Pull request by gambinish**
## **Description**

This PR updates the scam warning to be less aggressive and to also include a suggested ticker symbol.

I also took the opportunity to remove the redundant MATIC chain collision check, which is no longer needed since the remote chain list has been updated upstream.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26994?quickstart=1)

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/MMASSETS-384

## **Manual testing steps**

Add a custom network, with a custom/incorrect ticker (try adding Polygon with `MATIC` ticker)

Scam warning on token list page should reflect new copy from ticket.

## **Screenshots/Recordings**

Screenshot 2024-09-09 at 11 17 17 AM

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -3008,11 +3008,16 @@
     "message": "Edit network details"
   },
   "nativeTokenScamWarningDescription": {
-    "message": "This network doesn't match its associated chain ID or name. Many popular tokens use the name $1, making it a target for scams. Scammers may trick you into sending them more valuable currency in return. Verify everything before you continue.",
-    "description": "$1 represents the currency name"
+    "message": "The network currency symbol entered does not match it’s associated currency symbol on the current chain ID. You have entered $1 while the detected currency symbol should be $2. Please verify you are connected to the correct chain. ",
```
Comment (legobeat) [inline app/_locales/en/messages.json:3011]:
```suggestion
    "message": "The native token symbol does not match the expected symbol of the network with the associated chain ID. You have entered $1 while the expected token symbol is $2. Please verify you are connected to the correct chain.",
```

File: app/_locales/en/messages.json [hunk 2]
```diff
@@ -3008,11 +3008,16 @@
...
+    "description": "$1 represents the currency name, $2 represents the expected currency symbol"
+  },
+  "nativeTokenScamWarningDescriptionExpectedTokenFallback": {
+    "message": "something else",
+    "description": "graceful fallback for when token symbol isn't found"
   },
   "nativeTokenScamWarningTitle": {
-    "message": "This is a potential scam"
+    "message": "Incorrect Network Currency Symbol Entered",
```
Comment (legobeat) [inline app/_locales/en/messages.json:3019]:
```suggestion
    "message": "Unexpected Network Currency Symbol Entered",
```

File: app/_locales/en/messages.json (hunk 1 above)
Comment (gambinish) [reply app/_locales/en/messages.json:3011]: @alfeng6 Can you verify that the copy @legobeat is suggesting is approved by Product? I was going off of your direction in the [Jira ticket](https://consensyssoftware.atlassian.net/browse/MMASSETS-384). Looks good to me.

Comment (alfeng6) [reply app/_locales/en/messages.json:3011]: Copy looks good to me 👍

File: ui/components/app/assets/token-cell/token-cell.test.tsx [hunk 3]
```diff
@@ -90,7 +92,17 @@ describe('Token Cell', () => {
...
+  (useSelectorMock as jest.Mock).mockImplementation((selector) => {
+    if (selector === getTokenList) {
+      return MOCK_GET_TOKEN_LIST;
+    }
+    if (selector === getMultichainCurrentChainId) {
+      return '0x89';
+    }
+    return undefined;
+  });
+  // useSelector.mockReturnValue(MOCK_GET_TOKEN_LIST);
```
Comment (salimtb) [inline ui/components/app/assets/token-cell/token-cell.test.tsx:105]: is this comment relevant ? we should delete it

Comment (gambinish) [reply ui/components/app/assets/token-cell/token-cell.test.tsx:105]: woops, thanks for catching. will remove

Comment (gambinish) [reply ui/components/app/assets/token-cell/token-cell.test.tsx]: https://github.com/MetaMask/metamask-extension/pull/26994/commits/e6b567e22f7f3cb3227a752dde7f109db9b17266

File: ui/components/multichain/token-list-item/token-list-item.tsx [hunk 4]
```diff
@@ -100,6 +102,15 @@ export const TokenListItem = ({
   const metaMetricsId = useSelector(getMetaMetricsId);
   const isMetaMetricsEnabled = useSelector(getParticipateInMetaMetrics);
   const isMarketingEnabled = useSelector(getDataCollectionForMarketing);
+  const { safeChains } = useSafeChains();
+
+  const decimalChainId = isEvm && parseInt(hexToDecimal(chainId), 10);
+
+  const [safeChainDetails] = safeChains
```
Comment (darkwing) [inline ui/components/multichain/token-list-item/token-list-item.tsx:109]: Would `.find` work here instead?

Comment (legobeat) [reply ui/components/multichain/token-list-item/token-list-item.tsx:109]: Could also be worth expanding the condition from `safeChains` to `safeChains && typeof decimalChainId === 'number'` to skip for non-EVM chains?

Comment (gambinish) [reply ui/components/multichain/token-list-item/token-list-item.tsx]: https://github.com/MetaMask/metamask-extension/pull/26994/commits/74ebf605f41b8210cc20e15756a21dd5eb2daf21

**Review by legobeat — DISMISSED** [review]
LGTM

**Review by sahar-fehri — APPROVED** [review]

**Review by legobeat — APPROVED** [review]

**Review by vinnyhoward — APPROVED** [review]
LGTM!🟢

**Pull request merged**